File: goormlevel/159545/1. 0커플/0커플.cpp
#include <iostream>
#include <algorithm>

using namespace std;

int main() {
	ios_base::sync_with_stdio(false);
	cin.tie(nullptr);
	
	int n = 0;
	if (!(cin >> n)) return 0;

	int nlist[100001];
	for (int i = 0; i < n; i++) {
		if (!(cin >> nlist[i])) return 0;
	}

	sort(nlist, nlist + n);

	int answer = 0;
	for (int i = 0; i < n / 2; i++) {
		answer += nlist[i] + nlist[n - 1 - i];
	}

	cout << answer << "\n";

	return 0;
}

File: goormlevel/159545/1. 0커플/0커플.py
import sys

input = sys.stdin.readline
write = sys.stdout.write

def solve():
	n = int(input().rstrip())
	nlist = list(map(int, input().split()))

	nlist.sort()

	answer = 0
	for i in range(n // 2):
		answer += nlist[i] + nlist[n - 1 - i]

	write(str(answer) + "\n")

if __name__ == "__main__":
	solve()